uint64 handle
uint64 serial_number
string model_name
string firmware_version
